torch/jit/_recursive.py

```diff
@@ -230,6 +230,14 @@ def infer_type(name, item):
         # If we got here, this is a regular "data" attribute, Add it to the concrete type
         attr_type = infer_type(name, value)
         if attr_type is not None:
+            # If the attribute is a Class type and its name begins with "torch.jit.backends",
+            # it must be an instance of a lowered module.
+            if isinstance(attr_type, torch._C.ClassType):
+                prefix = (".").join(attr_type.str().split(".")[:3])
+                if prefix == "torch.jit.backends":
+                    sub_concrete_type = torch._C.ConcreteModuleType.from_jit_type(attr_type)
+                    concrete_type_builder.add_module(name, sub_concrete_type)
```

Comment:
Alright so this is clowntown but the root of the problem is that lowered modules are of type `torch._C.ScriptModule`, so they cannot be added as submodules of a module using `add_module` and get processed by the existing flow that works both for unscripted and scripted submodules.

I think the cleanest solution to this problem is to wrap the `torch._C.ScriptModule` in a `ScriptModule`, just like we do for regular, non-lowered scripted `Modules`. The one disadvantage here is that we will have to do this wrapping in C++ and there does not seem to be a precedent for that (for good reason I assume).

Reply:
Nvm I did it, hope you like it